src/library_ccall.js

```diff
@@ -31,6 +31,9 @@ mergeInto(LibraryManager.library, {
 #endif
       'string': (str) => {
         var ret = 0;
+         if (typeof str !== 'string') {
+    throw new TypeError('Argument must be a string');
+  }
```

Comment:
How about:

```
#if ASSERTIONS
  assert(typeof str == 'string');
#endif
```

And maybe at a test for this?  (or modify one of the existing ccall tests).

Reply:
That could work but showing the user the error they have made while giving an input would be more worthwhile and would guide the user in providing another prompt much more successfully.

Reply:
Sure, you can do `assert(typeof str == 'string', 'Argument must be a string');` if you want to include a message.

Reply:
Great! let me just implement it then.